code/mob/living/critter/flockcritter_parent.dm

```diff
@@ -536,3 +536,5 @@
 		amounttopay = min(F.resources, difference, 10)
 		F.pay_resources(amounttopay)
 		target.currentmats += amounttopay
+		if(F.resources > 0 && F.is_npc == 0) //npc check just to make sure it doesnt interfere with their ai.
```

Comment:
`== 0` or `!F.is_npc`?

also first `> 0` is useless

Reply:
i dont see how the `> 0` is useless?

Reply:
unless it can be negative numbers, you can just check F.resources

Reply:
oh right. oops